### ISSUE TITLE
Static GIF Images Not Displaying

### ISSUE THREAD

**Issue by dustinwilson** [tootsuite/mastodon#7997]
I've been having quite a bit of image issues lately it seems. OTL. I run a single user instance at [mastodon.dustinwilson.com](https://mastodon.dustinwilson.com) and recently migrated from a Docker install to a regular install. I am currently running v2.4.2.

I noticed this morning after viewing my feed on a boost from Gargron that static GIF images aren't displaying. [Here is the toot](https://mastodon.social/@timecowboy/100356334241445491). After some digging I decided to just force a redownload of the image and see what would happen:

```
cd ~/live
RAILS_ENV=production bundle exec rails c
m = MediaAttachment.find_by(remote_url: 'https://files.mastodon.social/media_attachments/files/004/913/346/original/7a7dc031509cb55b.gif')
m.file.destroy
m.save
```

When I refresh the page paperclip does its thing and places images in the folders but the images don't appear in the browser. This is where it gets weird. Here are the images on my server:

<https://mastodon.dustinwilson.com/system/media_attachments/files/000/031/505/small/a811ac97ce83941f.gif>
<https://mastodon.dustinwilson.com/system/media_attachments/files/000/031/505/original/a811ac97ce83941f.gif>

The "small" one is the actual original of the image while the "original" is an empty file. It only does this on gif images.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by dustinwilson** [comment]
Forgot to attach the log.

```
Jul 11 09:48:55 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] Trying to link /tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-lnzcie to /tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-1ye6w1c.gif
Jul 11 09:48:55 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] Command :: file -b --mime '/tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-1ye6w1c.gif'
Jul 11 09:48:55 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] Command :: identify -format %n '/tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-lnzcie'
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] Trying to link /tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-aapy9n.gif to /tmp/6859cfb7e357091a736a3187082b842320180711-8160-wvdmk1.gif
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] Command :: identify -format %n '/tmp/6859cfb7e357091a736a3187082b842320180711-8160-wvdmk1.gif'
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] Trying to link /tmp/8d777f385d3dfec8815d20f7496026dc20180711-8160-twf6mk.gif to /tmp/a9fbeed5765b10db0668022fe430042020180711-8160-tnyri4.gif
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] Trying to link /tmp/6859cfb7e357091a736a3187082b842320180711-8160-wvdmk1.gif to /tmp/0f5a9b630b66a842864a3e09d700cc2a20180711-8160-gkdzf4.gif
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] Command :: file -b --mime '/tmp/0f5a9b630b66a842864a3e09d700cc2a20180711-8160-gkdzf4.gif'
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] saving media_attachments/files/000/031/505/original/a811ac97ce83941f.gif
Jul 11 09:48:56 server bundle[8084]: [946c8148-e233-42bf-8afd-3fc2e3a85576] [paperclip] saving media_attachments/files/000/031/505/small/a811ac97ce83941f.gif
```

**Comment by hugogameiro** [comment]
Funny, I just saw that exact same image fail to load on another server also not running Docker and on my Docker install it worked as expected. Sorry for not introducing much into solving this but at least I can confirm that it is not just you and it's either a problem with non animated gif in general or that specific file.

**Comment by dustinwilson** [comment]
All gifs for me don't load, not just that one... but glad I'm not the only one with these issues (unlike the other stuff that's happened lately). Haha.

**Comment by hugogameiro** [comment]
After doing some more digging on this, I found the same issue on other non docker instances or non Object Storage instances. I would guess and say this issue has more to do with local storage than Docker but it's just a guess.
From what I could find it was working until December 4 of 2017 and on January 11 2018 it was not and all static gif present this problem.
Doing a quick look on the commits around that date I found this PR #6000 but it's just a guess and can't offer much help about the code pushed there.
